source/braille.py

```diff
@@ -2430,26 +2489,65 @@ def _get_shouldAutoTether(self) -> bool:
 	displaySize: int
 	_cache_displaySize = True
 
-	def _get_displaySize(self):
+	def _get_displaySize(self) -> int:
 		"""Returns the display size to use for braille output.
-		Handlers can register themselves to L{filter_displaySize} to change this value on the fly.
+		This is calculated from l{displayDimensions}.
+		Handlers can register themselves to L{filter_displayDimensions} to change this value on the fly.
 		Therefore, this is a read only property and can't be set.
 		"""
-		numCells = self.display.numCells if self.display else 0
-		currentDisplaySize = filter_displaySize.apply(numCells)
-		if self._displaySize != currentDisplaySize:
-			displaySizeChanged.notify(displaySize=currentDisplaySize)
-			self._displaySize = currentDisplaySize
-		return currentDisplaySize
+		numRows, numCols = self.displayDimensions
+		displaySize = numRows * numCols
+		# For backwards compatibility, we still set the internal cache.
+		self._displaySize = displaySize
+		return displaySize
 
 	def _set_displaySize(self, value):
 		"""While the display size can be changed while a display is connected
 		(for instance see L{brailleDisplayDrivers.alva.BrailleDisplayDriver} split point feature),
 		it is not possible to override the display size using this property.
-		Consider registering a handler to L{filter_displaySize} instead.
+		Consider registering a handler to L{filter_displayDimensions} instead.
 		"""
 		raise AttributeError(
-			f"Can't set displaySize to {value}, consider registering a handler to filter_displaySize",
+			f"Can't set displaySize to {value}, consider registering a handler to filter_displayDimensions",
+		)
+
+	displayDimensions: DisplayDimensions
+	_cache_displayDimensions = True
+
+	def _get_displayDimensions(self) -> DisplayDimensions:
+		rawDisplayDimensions = DisplayDimensions(
+			numRows=self.display.numRows if self.display else 0,
+			numCols=self.display.numCols if self.display else 0,
+		)
+		filteredDisplayDimensions = filter_displayDimensions.apply(rawDisplayDimensions)
+		# Would be nice if there were a more official way to find out if the displaySize filter is currently registered by at least 1 handler.
```

Comment:
It should be pretty trivial to extend extensionPoints.util with an additional property that returns `len(self._handlers)`.